include/sys/util_macro.h

```diff
@@ -620,6 +620,17 @@ extern "C" {
  */
 #define MACRO_MAP_CAT_N(N, ...) MACRO_MAP_CAT_N_(N, __VA_ARGS__)
 
+/**
+ * @brief Align a @p value up by to @p alignment
+ *
+ * Note, @p alignemnt should be a power-of-two
+ * 
+ * @param value the value to align up
+ * @param alignemnt the required alignemnt
+ * @return @p value aligned up to @p alignment
+ */
+#define ALIGN_UP(value, alignment) ((((uintptr_t)value) + (alignment)-1) & ~((alignment)-1))
```

Comment:
Two questions.

1. We currently have the ROUND_UP() macro that is defined in include/sys/util.h that does much the same thing.  Is the primary value of this macro based on that it uses ' uintptr_t' instead of 'unsigned long'?
2. Should 'alignment' be cast to 'uintptr_t'? I am thinking of a case where the caller passes in a signed char, or unsigned char. (I'll be honest ... I get mixed up on type promotion).

Reply:
Good question. Possibly intptr_t